src/bin/wasm2obj.rs

```diff
@@ -259,6 +266,7 @@ fn handle_module(
             CompilationStrategy::Auto | CompilationStrategy::Cranelift => {
                 Cranelift::compile_module(
                     &module,
+                    module_translation_state,
```

Comment:
@fitzgen, this isn't going to work because the `Compiler` trait expects four parameters here. It seems that the `ModuleTranslationState` (a cranelift-specific thing) shouldn't be exposed by the compiler-agnostic `Compiler` trait. So I looked at converting a `ModuleTranslationState` to `Module::signatures` but that doesn't seem quite right; the other thought is to just pass an empty `ModuleTranslationState::new()` to `FuncTranslator` elsewhere and avoid adding `ModuleTranslationState` to `ModuleTranslation` at the current time. Thoughts?

Reply:
I haven't really thought too much about this.

Any compiler will need its own equivalent of the type signatures data.

It may make sense to have it be an associated type that we can customize to each compiler's representation of shared module state that is required during function compilation:

```rust
trait Compiler {
    type ModuleState;

    fn compile_module(state: &ModuleState, ...);
}
```